### PR TITLE
Added User profile feature

### PR DESCRIPTION
**Pull request by Ayush0730**
## What is the change?
Added User profile feature, profile image will be shown on user profile page and dashboard, also added feature to edit profile image

## Related issue?
close: #512 

## How was it tested?
Tested Locally

## Checklist:
Before you create this PR, confirm all the requirements listed below by checking the checkboxes `[x]`:

- [x] Have you followed the [Contribution Guidelines](https://github.com/ALPHAVIO/BlogSite/blob/master/CONTRIBUTING.md) while contributing.
- [x] Have you checked there aren't other open [Pull Requests](https://github.com/ALPHAVIO/BlogSite/pulls) for the same update/change?
- [] Did you lint your code before making the Pull Request?
- [] Have you made corresponding changes to the documentation?
- [x] Your submission doesn't break any existing feature.
- [x] Have you tested the code before submission?

## Screenshots or Video:

https://user-images.githubusercontent.com/68139755/115528020-42e1c180-a2af-11eb-88f9-3a5f40cbfb71.mp4

### CODE REVIEW

**Comment by Ayush0730** [comment]
To clear any doubts, at 0:44, the profile picture shown is default for everyone, if they do not upload any profile picture at the time of signup. And users can see other users' profile picture

**Comment by Ayush0730** [comment]
Improved edit profile page

image

File: views/useritems/dashboard.ejs [hunk 1]
```diff
@@ -6,6 +6,8 @@
         <div class="col-lg-11 card bg-light mx-2 mx-lg-auto card p-0">      
             <%# Display author's name %> 
             <div class="card-header text-center">
+                <%# Display author's profile %>
+                <img src="<%= user.photo %>" class="img-thumbnail mx-auto" width="200px" alt="User Profile Image">
```
Comment (Pranav016) [inline views/useritems/dashboard.ejs:10]: Don't add inline styling
replace `px` with one of these `rem em vh vw %` for responsiveness
add some border radius

**Review by Pranav016 — CHANGES_REQUESTED** [review]
Looks good

**Review by Pranav016 — APPROVED** [review]

**Comment by Pranav016** [comment]
resolve conflicts **carefully**, after that I can merge

**Comment by Pranav016** [comment]
/gssoc/

**Comment by Ayush0730** [comment]
> resolve conflicts **carefully**, after that I can merge

Please check if it is Ok

**Comment by Pranav016** [comment]
> > resolve conflicts **carefully**, after that I can merge
> 
> Please check if it is Ok

The CI run and some changes were requested by the GitHub bot..

**Comment by Ayush0730** [comment]
> > > resolve conflicts **carefully**, after that I can merge
> > 
> > 
> > Please check if it is Ok
> 
> The last commit doesn't show anything so I am hoping the conflicts were resolved correctly otherwise we'll correct the bug later (if any), no problem.
> 
> Also, after resolving the conflicts the CI run and some changes were requested by the GitHub bot that need to be resolved
> Then I can merge.

How to solve that?

**Comment by Pranav016** [comment]
> > > > resolve conflicts **carefully**, after that I can merge
> > > 
> > > 
> > > Please check if it is Ok
> > 
> > 
> > The last commit doesn't show anything so I am hoping the conflicts were resolved correctly otherwise we'll correct the bug later (if any), no problem.
> > Also, after resolving the conflicts the CI run and some changes were requested by the GitHub bot that need to be resolved
> > Then I can merge.
> 
> How to solve that?

Run command `prettier --write .` locally.
Then run `npm run lint` to check if it is still giving any error..

**Comment by Ayush0730** [comment]
> > > > > resolve conflicts **carefully**, after that I can merge
> > > > 
> > > > 
> > > > Please check if it is Ok
> > > 
> > > 
> > > The last commit doesn't show anything so I am hoping the conflicts were resolved correctly otherwise we'll correct the bug later (if any), no problem.
> > > Also, after resolving the conflicts the CI run and some changes were requested by the GitHub bot that need to be resolved
> > > Then I can merge.
> > 
> > 
> > How to solve that?
> 
> Run command `prettier --write .` locally.
> Then run `npm run lint` to check if it is still giving any error..

image
😥

**Comment by Ayush0730** [comment]
after running npm run lint command it has done many file changes

should i push the changes?

**Comment by Pranav016** [comment]
> after running npm run lint command it has done many file changes
> 
> should i push the changes?

Also do `npm run lint-check` to check if all errors are gone

**Comment by Pranav016** [comment]
> > > > > > resolve conflicts **carefully**, after that I can merge
> > > > > 
> > > > > 
> > > > > Please check if it is Ok
> > > > 
> > > > 
> > > > The last commit doesn't show anything so I am hoping the conflicts were resolved correctly otherwise we'll correct the bug later (if any), no problem.
> > > > Also, after resolving the conflicts the CI run and some changes were requested by the GitHub bot that need to be resolved
> > > > Then I can merge.
> > > 
> > > 
> > > How to solve that?
> > 
> > 
> > Run command `prettier --write .` locally.
> > Then run `npm run lint` to check if it is still giving any error..
> 
> image
> 😥

Do you have up to date project? Because a lot of these errors I solved a couple days ago

**Comment by Pranav016** [comment]
> after running npm run lint command it has done many file changes
> 
> should i push the changes?

Yes you can push it

**Comment by Ayush0730** [comment]
> > after running npm run lint command it has done many file changes
> > should i push the changes?
> 
> Yes you can push it

Its all messed up

**Comment by Pranav016** [comment]
> Its all messed up

still showing errors in the CI, I'll check

**Comment by Pranav016** [comment]
> > Its all messed up
> 
> still showing errors in the CI, I'll check

on my machine it's not showing any linting errors.
`npm run lint` should solve the errors, idk whats the issue here

**Comment by Ayush0730** [comment]
> > > Its all messed up
> > 
> > 
> > still showing errors in the CI, I'll check
> 
> on my machine it's not showing any linting errors.
> `npm run lint` should solve the errors, idk whats the issue here

What should I do now?

**Comment by Pranav016** [comment]
/level2/

**Pull request merged**